common/str.cpp

```diff
@@ -40,18 +40,18 @@ void lockMemoryPoolMutex() {
 	if (!g_system || !g_system->backendInitialized())
 		return;
 	if (!g_refCountPoolMutex)
-		g_refCountPoolMutex = g_system->createMutex();
-	g_system->lockMutex(g_refCountPoolMutex);
+		g_refCountPoolMutex = new Mutex();
+	g_refCountPoolMutex->lock();
 }
 
 void unlockMemoryPoolMutex() {
 	if (g_refCountPoolMutex)
-		g_system->unlockMutex(g_refCountPoolMutex);
+		g_refCountPoolMutex->unlock();
 }
 
 void String::releaseMemoryPoolMutex() {
```

Comment:
The Deep Code is right here. We never ever call this method...

Reply:
By "this method" I guess you mean `releaseMemoryPoolMutex()`? This is called from `OSystem::destroy()`.
The idea is that if we create a `String` during the destruction of the `OSystem` instance (and we do), it should not be using the mutex as this would crash. So we delete and clear the mutex before doing a `delete this` in `OSystem::destroy()`.

Reply:
Ah, right. I overlooked it somehow.